### PR TITLE
Update six version to 1.12.0 and fix legacy_create_init issue

### PR DESCRIPTION
**Pull request by aaliddell**
When the @six//:six library is used on a target without the `legacy_create_init` flag disabled, the library will not be usable due to __init__.py being empty and the six code will be in six.py. This change forces six to occupy the __init__ name, preventing this file getting created regardless of the `legacy_create_init` setting.

See comments on a74c43bbd906a42b6bf70668c5d515c6f1d8d2cb for context.

### CODE REVIEW

**Comment by laszlocsomor** [comment]
Hey, any updates on this?

The [protobuf project on Bazel CI](https://buildkite.com/bazel/protobuf/builds) is still broken, so we won't know if Bazel can still build protobuf.

File: WORKSPACE [hunk 1]
```diff
@@ -14,9 +14,9 @@ new_local_repository(
 
 http_archive(
     name = "six",
-    build_file = "@//:six.BUILD",
-    sha256 = "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a",
-    urls = ["https://pypi.python.org/packages/source/s/six/six-1.10.0.tar.gz#md5=34eed507548117b2ab523ab14b2f8b55"],
+    build_file = "@//:third_party/six.BUILD",
+    sha256 = "d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73",
+    urls = ["https://pypi.python.org/packages/source/s/six/six-1.12.0.tar.gz"],
 )
```
Comment (Yannic) [inline WORKSPACE]: I don't think this `http_archive` is required in `WORKSPACE`, it's already loaded by `protobuf_deps()` a few lines above.

Comment (aaliddell) [reply WORKSPACE]: I've just left the existing declaration locations as-is, but it'd be worth considering removing the duplicate.

Comment (aaliddell) [reply WORKSPACE]: Added commit

**Comment by acozzette** [comment]
@aaliddell Thank you for fixing this. Could you merge master (or rebase) again? It looks like the CI tests won't run if the PR can't be merged automatically.

**Comment by aaliddell** [comment]
Rebase done

**Comment by aaliddell** [comment]
The test failures again do not look related to the changes in this PR

**Pull request merged**